### ISSUE TITLE
Add tests for v2.1.0 Transaction Requests

### ISSUE THREAD

**Issue by simonredfern** [OpenBankProject/OBP-API#223]
Copy from v2.0.0 Transaction Requests tests and modify to handle new details structure

**Comment by hongwei1** [comment]
Fixed some parts to make it work on the V2.1.0

**Comment by hongwei1** [comment]
Hi Simon,
I have problems on the test for "SEPA" and "FREE_FORM". It confused me a lot .

* 1 
I think these features have not been implemented.
As the following code. They can not use "createTransactionRequestv210" ,because  of the "Empty" parameters .

```scala
case "SEPA" => {
                  Connector.connector.vend.createTransactionRequestv210(u, fromAccount, Empty, transactionRequestType, transDetails, transDetailsSerialized)
                }
case "FREE_FORM" => {
                  Connector.connector.vend.createTransactionRequestv210(u, fromAccount, Empty, transactionRequestType, transDetails, transDetailsSerialized)
                }
```

* 2 
 "SEPA" and "FREE_FORM" have not enough input to store in the database .
As the following code ,they lack Bank_Id or Count_id .

```scala
case class TransactionRequestDetailsSEPAJSON(
              value : AmountOfMoneyJSON,
               IBAN: String,
               description : String
                ) extends TransactionRequestDetailsJSON

case class TransactionRequestDetailsFreeFormJSON(
                value : AmountOfMoneyJSON
                ) extends TransactionRequestDetailsJSON
```

*3 My Solution
- Change the format  TransactionRequestDetailsSEPAJSON and TransactionRequestDetailsFreeFormJSON to the same as TransactionRequestDetailsSandBoxTanJSON .

- Write new Mapper methods to save the different format.

what is your option? Any good solutions ?
